api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceService.java

```diff
@@ -21,6 +22,8 @@
 
   DbWorkspace findByWorkspaceId(long workspaceId);
 
+  Optional<DbWorkspace> getWorkspaceMaybe(long workspaceId);
```

Comment:
You could argue that `findByWorkspaceId()` should be renamed `findByWorkspaceIdOrThrow()` and the optional-returning variant should have a simpler name like `findActiveWorkspace()`.

Reply:
I am OK with using `Optional`s but having to suffix all of our function names with `Maybe` seems like a bit much. I think it will be easy to distinguish an optional from not since the type signature is so different.

Reply:
Optional is the next best thing to Kotlin. I don't understand why you're only OK with it.

Except you can't overload based on return type.
![Screen Shot 2019-11-08 at 5 04 27 PM](https://user-images.githubusercontent.com/53479492/68513792-eec55780-0249-11ea-98ea-28057c8003a4.png)

My preference is to name the new optional-returning one `findActiveWorkspaceById()` or `findWorkspaceByIdIfActive` and the old one `findActiveWorkspaceByIdOrThrow()`. In other words, returning optional for something that frequently won't be there should be the norm, and throwing should be really exceptional (no pun intended).

Reply:
I like the additional safety it brings but the tradeoff, to me, is that I find the interface for using it to be a bit clunky and verbose.